collectors/chain_collector.go

```diff
@@ -14,11 +15,17 @@ type ChainCollector struct {
 	syncedToChain      *prometheus.Desc
 
 	lnd lndclient.LightningClient
+
+	// errChan is a channel that we send any errors that we encounter into.
+	// This channel should be buffered so that it does not block sends.
+	errChan chan<- error
```

Comment:
👍 

At times I update my `lnd` node, which causes `lndmon` to just hang rather than shutdown and re-establish connectivity like it should.